types/host.go

```diff
@@ -63,17 +63,20 @@ type VMStat interface {
 
 // HostInfo contains basic host information.
 type HostInfo struct {
-	Architecture      string    `json:"architecture"`            // Hardware architecture (e.g. x86_64, arm, ppc, mips).
-	BootTime          time.Time `json:"boot_time"`               // Host boot time.
-	Containerized     *bool     `json:"containerized,omitempty"` // Is the process containerized.
-	Hostname          string    `json:"name"`                    // Hostname
-	IPs               []string  `json:"ip,omitempty"`            // List of all IPs.
-	KernelVersion     string    `json:"kernel_version"`          // Kernel version.
-	MACs              []string  `json:"mac"`                     // List of MAC addresses.
-	OS                *OSInfo   `json:"os"`                      // OS information.
-	Timezone          string    `json:"timezone"`                // System timezone.
-	TimezoneOffsetSec int       `json:"timezone_offset_sec"`     // Timezone offset (seconds from UTC).
-	UniqueID          string    `json:"id,omitempty"`            // Unique ID of the host (optional).
+	Architecture  string    `json:"architecture"`            // Hardware architecture (e.g. x86_64, arm, ppc, mips).
+	BootTime      time.Time `json:"boot_time"`               // Host boot time.
+	Containerized *bool     `json:"containerized,omitempty"` // Is the process containerized.
+	Hostname      string    `json:"name"`                    // Hostname
+	// FQDN is the Fully qualified domain name, for linux and darwin it requires CGO.
+	// if CGO is not enabled it'll fallback to os.Hostname(), the same as Hostname.
+	FQDN              string   `json:"fqdn"`
```

Comment:
As a user of the library I would prefer the library indicate when it cannot collect the FQDN rather than silently fallback. IMO the most clear way to accomplish this would be to use `*string` and leave it nil when the data is unavailable. Then document similar to what you have that not a provider implementations support this field and it may be nil.

Reply:
done, I added Domain. Now if Domain cannot be retrieved, Cgo is disabled for linux and darwin, FQDN will be empty as well as Domain.